zerver/views/users.py

```diff
@@ -347,27 +338,36 @@ def get_stream_name(stream: Optional[Stream]) -> Optional[str]:
 
 
 @require_member_or_admin
-@has_request_variables
+@typed_endpoint
 def patch_bot_backend(
     request: HttpRequest,
     user_profile: UserProfile,
     bot_id: int,
-    full_name: Optional[str] = REQ(default=None),
-    role: Optional[int] = REQ(
-        default=None,
-        json_validator=check_int_in(
-            UserProfile.ROLE_TYPES,
-        ),
-    ),
-    bot_owner_id: Optional[int] = REQ(json_validator=check_int, default=None),
-    config_data: Optional[Dict[str, str]] = REQ(
-        default=None, json_validator=check_dict(value_validator=check_string)
-    ),
-    service_payload_url: Optional[str] = REQ(json_validator=check_url, default=None),
-    service_interface: int = REQ(json_validator=check_int, default=1),
-    default_sending_stream: Optional[str] = REQ(default=None),
-    default_events_register_stream: Optional[str] = REQ(default=None),
-    default_all_public_streams: Optional[bool] = REQ(default=None, json_validator=check_bool),
+    *,
+    full_name: Optional[str] = None,
+    role: Optional[
+        Json[
+            Annotated[
+                int,
+                AfterValidator(
+                    lambda x: check_int_param_in(
+                        "role",
+                        x,
+                        UserProfile.ROLE_TYPES,
+                    )
+                ),
+            ]
+        ]
+    ] = None,
```

Comment:
Any ideas on a better way to do this `role` parameter without so much nesting? I suppose an option would be to just declare a `RoleParameterType` and use that here? I think we have some other views that use a lot of enumerated integer parameters, so it'd be good to have a clean pattern for these.

Reply:
https://docs.pydantic.dev/2.0/usage/types/number_types/ might offer some inspiration.

Reply:
Creating `RoleParameterType` is an option, I didn't create it initially because `role` isn't being used in any endpoint in any other file, but its being used twice here so it might be better to create a type. 

We could use `Enums` or `Literal` but that would sacrifice the ability to pass a list. Both those would require us to hardcode specific values and do not take lists. `Literal` does take tuples but passing tuples throws linting errors.

Reply:
Yeah I think making a type for use in this file is worth it; if nothing else it'd be a good example pattern to try.